src/SequentialEventBus.ts

```diff
@@ -1,23 +1,36 @@
-import { ISequentialEventListener, IEvent, IEventConstructor } from './ISequentialEventListener';
+import type { IEvent, ISequentialEventListener } from './ISequentialEventListener';
+import type { Type } from '@nestjs/common';
+import { ModuleRef } from '@nestjs/core';
+import { SEQUENTIAL_EVENT, SEQUENTIAL_EVENT_LISTENER } from './constants';
+import 'reflect-metadata';
+import { Injectable } from '@nestjs/common';
 
 /**
  * The main difference from the Nest EventBus is that event handlers must finish its work to proceed.
+ *
+ * Implementation for binding events based on https://github.com/nestjs/cqrs/blob/master/src/event-bus.ts
  */
+@Injectable()
 export class SequentialEventBus {
-  private listeners: Map<IEventConstructor, ISequentialEventListener[]> = new Map();
+  private listeners: Map<string, ISequentialEventListener[]> = new Map();
```

Comment:
why uuid and string if you can assign the same listener to many events without any refactor?